### PR TITLE
[Rest Server] Remove jobName in config file

### PR DESCRIPTION
**Pull request by abuccts**
Remove `jobName` in config file which is duplicate with the name in request path.

### CODE REVIEW

File: rest-server/src/routes/job.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ router.route('/:jobName')
     .get(jobController.get)
 
     /** PUT /api/v1/jobs/:jobName - Update job */
-    .put(tokenConfig.check, param.validate(jobConfig.schema), jobController.update)
+    .put(tokenConfig.check, param.validate(jobConfig.schema, {'params': ['jobName']}), jobController.update)
```
Comment (hwuu) [inline rest-server/src/routes/job.js]: As we discussed yesterday, please use a separate middleware to synthesis req.body.

File: job-tutorial/README.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ Below please find the detailed explanation for each of the parameters in the con
 
 | Field Name                     | Schema                     | Description                              |
 | :----------------------------- | :------------------------- | :--------------------------------------- |
-| `jobName`                      | String in `^[A-Za-z0-9\-._~]+$` format, required | Name for the job, need to be unique |
+| `jobName`                      | String in `^[A-Za-z0-9\-._~]+$` format, optional | Name for the job, need to be unique |
```
Comment (hwuu) [inline job-tutorial/README.md]: Can we mark this field ‘deprecated’, instead of ‘optional’ here?

**Comment by fanyangCS** [comment]
let's not change json file, change the rest api instead.

**Comment by abuccts** [comment]
pls refer to #398

**Pull request closed without merging**